src/libfsm/endids.c

```diff
@@ -129,7 +129,7 @@ fsm_endid_free(struct fsm *fsm)
 static unsigned long
 hash_state(fsm_state_t state)
 {
-	return fsm_hash_id(state);
+	return (unsigned long)fsm_hash_id(state);
```

Comment:
`fsm_hash_id()` returns `uint64_t`. the minimum maximum for `unsigned long` is 32-bit. this'll bite us when compiling libfsm to wasm. since you're introducing a change around these variables anyway, let's make all these hash variables uint64_t now

Reply:
Sure, will do, and intentionally using `uint64_t` rather than `unsigned long` will remove some other annoyances elsewhere.

Reply:
Updated in a128cb27756bcf505b1a87eef7cb2a5ea82b8766.